src/api/controllers/messages.ts

```diff
@@ -22,16 +22,31 @@ export async function handleInternalMessageCreate(data: DiscordPayload) {
   if (payload.member && guild) {
     // If in a guild cache the author as a member
     await structures.createMember(
-      { ...payload.member, user: payload.author },
+      {
+        ...payload.member,
+        nick: payload.member.nick === undefined ? null : payload.member.nick,
+        roles: payload.member.roles || [],
+        joined_at: payload.member.joined_at || "",
+        deaf: payload.member.deaf || false,
+        mute: payload.member.mute || false,
+        user: payload.author,
+      },
       guild.id,
     );
   }
 
   payload.mentions.forEach((mention) => {
     // Cache the member if its a valid member
-    if (mention.member && guild) {
+    if (mention && guild) {
       structures.createMember(
-        { ...mention.member, user: mention },
+        {
+          nick: mention.nick === undefined ? null : mention.nick,
+          roles: mention.roles || [],
+          joined_at: mention.joined_at || "",
+          deaf: mention.deaf || false,
+          mute: mention.mute || false,
+          user: mention,
+        },
```

Comment:
Irrelevant changes, please remove them, maybe create a new PR fixing all of these.